src/cr/cube/matrix/cubemeasure.py

```diff
@@ -914,14 +918,31 @@ def factory(cls, cube, dimensions, slice_idx):
             else _CatXCatWeightedCubeCounts
         )
         return WeightedCubeCountsCls(
-            dimensions, counts[cls._slice_idx_expr(cube, slice_idx)], diff_nans
+            dimensions,
+            counts[cls._slice_idx_expr(cube, slice_idx)],
+            diff_nans,
+            dimensions_are_array,
         )
 
     @lazyproperty
     def column_bases(self):
         """2D np.float64 ndarray of column-proportion denominator for each cell."""
         return np.broadcast_to(self.columns_margin, self.weighted_counts.shape)
 
+    @lazyproperty
+    def column_comparable_counts(self):
+        """2D np.float64 ndarray of weighted-count for cells in columns that share a base
+
+        Cells in a column don't share a base when they go across array subvariables.
+        Raises when counts are not comparable.
+        """
+        if self._dimensions_are_array[1]:
```

Comment:
oh nice. The negative indexing still feels weird to me (maybe just because it's different between python and R), so it felt better to add  `_row/column_is_array` property.

Reply:
oh nevermind, I was confused about the difference between the dimensions on `_Cube` and on `_Slice`. So my code is like you suggested.